File: apps/query-burst/src/main/java/com/booster/queryburst/order/web/dto/response/OrderSummaryResponse.java
package com.booster.queryburst.order.web.dto.response;

import com.booster.queryburst.order.application.dto.OrderSummaryResult;
import com.booster.queryburst.order.domain.OrderStatus;

import java.time.LocalDateTime;

public record OrderSummaryResponse(
        Long orderId,
        Long memberId,
        String memberName,
        OrderStatus status,
        Long totalAmount,
        LocalDateTime orderedAt
) {
    public static OrderSummaryResponse from(OrderSummaryResult result) {
        return new OrderSummaryResponse(
                result.orderId(),
                result.memberId(),
                result.memberName(),
                result.status(),
                result.totalAmount(),
                result.orderedAt()
        );
    }
}

File: apps/kotlin/shopping-service/src/main/kotlin/com/booster/kotlin/shoppingservice/order/web/dto/response/OrderSummaryResponse.kt
package com.booster.kotlin.shoppingservice.order.web.dto.response

import com.booster.kotlin.shoppingservice.order.domain.Order
import com.booster.kotlin.shoppingservice.order.domain.OrderStatus
import java.time.LocalDateTime

data class OrderSummaryResponse(
    val orderId: Long,
    val status: OrderStatus,
    val totalPrice: Long,
    val itemCount: Int,
    val createdAt: LocalDateTime,
) {
    companion object {
        fun from(order: Order) = OrderSummaryResponse(
            orderId = order.id,
            status = order.status,
            totalPrice = order.totalPrice,
            itemCount = order.items.size,
            createdAt = order.createdAt,
        )
    }
}
